### ISSUE TITLE
Unclear error displayed if user tries to pay IOU for deleted account request - reported by @Tushu17

### ISSUE THREAD

**Issue by mvtglobally** [Expensify/App#7435]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Request money from Account A to account B
2. Delete account A
3. Now login as Account B and click the pay button.

## Expected Result:
User should get a clear error why money can't be paid or request should be removed 

## Actual Result:
Unclear error displayed and user is unable to pay 

## Workaround:
unknown 

## Platform:

Where is this issue occurring?

- Web
- iOS
- Android
- Desktop App
- Mobile Web

**Version Number:**
**Reproducible in staging?:**
**Reproducible in production?:**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation 

https://user-images.githubusercontent.com/43995119/151295657-ec6f240f-5545-4583-ab34-74d612a37f42.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @Tushu17
**Slack conversation:** https://expensify.slack.com/archives/C01GTK53T8Q/p1642285631499100

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @mateocole (`AutoAssignerTriage`), see https://stackoverflow.com/c/expensify/questions/4749 for more details.

**Comment by mateocole** [comment]
Question on this, do we allow Account A to close if their are pending payments?

**Comment by Tushu17** [comment]
@mateocole Yes, We allow usersto delete their account even if they have pending payments.

**Comment by MelvinBot** [comment]
This issue has not been updated in over ***14*** days. @mateocole eroding to ***Weekly*** issue.

**Comment by MelvinBot** [comment]
Triggered auto assignment to @deetergp (`Engineering`), see https://stackoverflow.com/c/expensify/questions/4319 for more details.

**Comment by deetergp** [comment]
Not sure how this flew under the radar, but it can be worked on by an external contributor.

**Comment by MitchExpensify** [comment]
[Exported](https://www.upwork.com/jobs/~0144b990cad5fc395a) to Upwork and @Tushu17 invited for eventual reporting bonus payment

**Comment by Tushu17** [comment]
@MitchExpensify Sorry I almost forgot and applied to the job. Actually there's a similar issue(#7420) for this and I think this will get fixed there, so I think we should hold this one if it doesn't get fixed we can remove the hold.

**Comment by MitchExpensify** [comment]
Sounds like a plan @Tushu17

**Comment by MitchExpensify** [comment]
Holding on https://github.com/Expensify/App/issues/7420 to see if it fixes this issue once deployed

**Comment by deetergp** [comment]
Not overdue, just holding

**Comment by deetergp** [comment]
Still holding…

**Comment by rushatgabhane** [comment]
Unassigning myself because I'm OOO until May 14

**Comment by deetergp** [comment]
Same status as the last. Once @stitesExpensify sorts out https://github.com/Expensify/App/issues/7420, we'll re-test this one to see if it's fixed.

**Comment by MitchExpensify** [comment]
Waiting on #7420 to re-test

**Comment by MitchExpensify** [comment]
Same as above

**Comment by deetergp** [comment]
Still on hold

**Comment by deetergp** [comment]
How long are we going to hold on this one?

**Comment by stitesExpensify** [comment]
We're holding until I do [this issue](https://github.com/Expensify/App/issues/7420), which is currently very low priority.

**Comment by deetergp** [comment]
Still holding. Since @stitesExpensify's blocking issue is low priority, going to set this to monthly.

**Comment by MitchExpensify** [comment]
Still holding on https://github.com/Expensify/App/issues/7420

**Comment by deetergp** [comment]
No updates as yet.

**Comment by deetergp** [comment]
Not overdue

**Comment by deetergp** [comment]
Still on hold

**Comment by tjferriss** [comment]
Following instructions for the [weekly update chore](https://stackoverflow.com/c/expensify/questions/8208):

@deetergp this is one of the oldest issues in the /App repo. To help us clear out the large backlog of bugs, can you:
- Decide whether any proposals currently meet our guidelines and can be approved as-is
- For any that can't, please take this issue internal and treat it as one of your highest priorities
- If you have any questions, don't hesitate to start a discussion in #bug-zero

I recognize this issue is on hold so maybe the above is irrelevant, but once it's not on hold do you expect we should take this internal given the issue's age?

**Comment by deetergp** [comment]
Actually, it looks like the issue this was holding on has been merged, paid, and closed. Going to take it off hold and request another re-test from @mvtglobally in #qa and see if it's still broken.

**Comment by mvtglobally** [comment]
@deetergp 
I am getting slightly different error now, but still an error without clear explanation

https://user-images.githubusercontent.com/43995119/204436211-0fccac45-7a22-4e66-b9bd-de4ebcfbfd46.mp4

**Comment by MitchExpensify** [comment]
> Actually, it looks like the issue this was holding on has been merged, paid, and closed. Going to take it off hold and request another re-test from @mvtglobally in #qa and see if it's still broken.

Updated the title to reflect [this](https://github.com/Expensify/App/issues/7435#issuecomment-1329876665)

**Comment by deetergp** [comment]
I'm just going to remove external and take this one on.

**Comment by JmillsExpensify** [comment]
Awesome, sounds great! This is the second-oldest open issue in the repo without a PR. Now's a good time to close the loop on it.

**Comment by deetergp** [comment]
On my first few tries to recreate this, I got different results, but am now able to recreate @mvtglobally's results. I'm also noticing a _not-particuarly-useful-for-end-users_ error in the middle of the screen when User B navigates to their convo with User A after the latter has closed their account.

**Comment by JmillsExpensify** [comment]
Wowsers, that is indeed a not very useful RBR error message. We need to fail more gracefully here.

**Comment by MitchExpensify** [comment]
@Tushu17 Invited you to the new [Upwork Job](https://www.upwork.com/jobs/~01d2be9076770309b0) for reporting bonus payment

**Comment by JmillsExpensify** [comment]
Thanks so much for your honesty! @MitchExpensify With that in mind, I think we're set to close this issue out, right?

**Comment by MitchExpensify** [comment]
Yes @JmillsExpensify! Closing. And thanks @Tushu17, much appreciated
